### PR TITLE
Use case onboarding

### PR DESCRIPTION
**Pull request by neallred**
#### Summary
* Add `UseCaseOnboarding` feature flag.
* Add status for whether first admin onboarding has been completed.
* Add in-app migration for whether existing installations completed first admin onboarding. The current logic is simple: if there are any teams and >= 10 chats, consider it a completed installation.

#### Ticket Link
https://community-daily.mattermost.com/boards/workspace/fpbb7orqpiyfxqgg6fcu91y5xo/b6ropkdiowj8eugbt8kifrew79h/vptcho591nfry8rgfygxe8f7tbr/cyg1wqa6j97gat86c6ursd7hcqa

#### Release Note

```release-note
* Add `UseCaseOnboarding` feature flag.
* Add status for whether first admin onboarding has been completed.
* Add in-app migration for whether existing installations completed first admin onboarding.
```

### CODE REVIEW

File: app/migrations.go [hunk 1]
```diff
@@ -434,6 +435,50 @@ func (s *Server) doPlaybooksRolesCreationMigration() {
 
 }
 
+// arbitrary choice, though if there is an longstanding installation with less than 10 messages
+// putthing the first admin through onboarding shouldn't be very disruptive
```
Comment (mickmister) [inline app/migrations.go:439]:
```suggestion
// arbitrary choice, though if there is an longstanding installation with less than 10 messages,
// putting the first admin through onboarding shouldn't be very disruptive
```

**Review by mickmister — APPROVED** [review]
LGTM :tada:

**Comment by Willyfrog** [comment]
@hanzei can you please prioritize reviewing this PR as it is part of the Growth Spike?

File: api4/system.go [hunk 2]
```diff
@@ -903,6 +937,16 @@ func completeOnboarding(c *Context, w http.ResponseWriter, r *http.Request) {
...
 
+	firstAdminCompleteSetupObj := model.System{
+		Name:  model.SystemFirstAdminCompleteSetup,
+		Value: "true",
+	}
+
+	if err := c.App.Srv().Store.System().SaveOrUpdate(&firstAdminCompleteSetupObj); err != nil {
+		c.Err = model.NewAppError("setFirstAdminCompleteSetup", "api.error_set_first_admin_complete_setup", nil, err.Error(), http.StatusInternalServerError)
+		return
+	}
```
Comment (hanzei) [inline api4/system.go:948]: Let's keep all the business logic in the `app` layer and move code into `app.CompleteOnboarding`

File: api4/system.go [hunk 3]
```diff
@@ -68,6 +69,7 @@ func (api *API) InitSystem() {
 	api.BaseRoutes.System.Handle("/notices/{team_id:[A-Za-z0-9]+}", api.APISessionRequired(getProductNotices)).Methods("GET")
 	api.BaseRoutes.System.Handle("/notices/view", api.APISessionRequired(updateViewedProductNotices)).Methods("PUT")
 	api.BaseRoutes.System.Handle("/support_packet", api.APISessionRequired(generateSupportPacket)).Methods("GET")
+	api.BaseRoutes.System.Handle("/first_admin_setup", api.APISessionRequired(getFirstAdminCompleteSetup)).Methods("GET")
 	api.BaseRoutes.System.Handle("/onboarding/complete", api.APIHandler(completeOnboarding)).Methods("POST")
```
Comment (hanzei) [inline api4/system.go:73]: I forgot to add the session handler as part of my PR. Mind adding it here?

```suggestion
	api.BaseRoutes.System.Handle("/first_admin_setup", api.APISessionRequired(getFirstAdminCompleteSetup)).Methods("GET")
	api.BaseRoutes.System.APISessionRequired("/onboarding/complete", api.APIHandler(completeOnboarding)).Methods("POST")
```

Comment (hanzei) [inline api4/system.go:73]: 2/5 that the API endpoints should be more similar. It note clear a the reader that they belong to the same feature. Maybe we simplify to `/onboarding` `POST` and `GET`?

File: api4/system.go [hunk 4]
```diff
@@ -881,6 +883,38 @@ func updateViewedProductNotices(c *Context, w http.ResponseWriter, r *http.Reque
...
+			firstAdminCompleteSetupObj = &model.System{
+				Name:  model.SystemFirstAdminCompleteSetup,
+				Value: "false",
+			}
+		default:
+			c.Err = model.NewAppError("getFirstAdminCompleteSetup", "api.error_get_first_admin_complete_setup", nil, err.Error(), http.StatusInternalServerError)
+
+			return
+		}
+	}
```
Comment (hanzei) [inline api4/system.go:910]: 0/5 should this be it's own store method? Or at least an `app` method

**Review by hanzei — CHANGES_REQUESTED** [review]
Thanks for continuing my PR :+1: 

A few comments below

File: app/migrations.go [hunk 5]
```diff
@@ -434,6 +435,50 @@ func (s *Server) doPlaybooksRolesCreationMigration() {
 
 }
 
+// arbitrary choice, though if there is an longstanding installation with less than 10 messages,
+// putting the first admin through onboarding shouldn't be very disruptive
```
Comment (hanzei) [inline app/migrations.go:439]:
```suggestion
// putting the first admin through onboarding shouldn't be very disruptive.
```

**Review by hanzei — APPROVED** [review]
LGTM :+1:

**Pull request merged**